File: app/src/main/java/com/imangazalievm/bubbble/di/modules/ApplicationModule.java
package com.imangazalievm.bubbble.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private final Context context;

    public ApplicationModule(Context context) {
        this.context = context.getApplicationContext();
    }

    @Provides
    @Singleton
    Context provideApplicationContext() {
        return context;
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/di/global/ApplicationModule.kt
package com.imangazalievm.bubbble.di.global

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(private val context: Context) {

    @Provides
    @Singleton
    fun provideApplicationContext(): Context = context

}